homework24Networking/src/ba/bitcamp/homework24Networking/Server.java

```diff
@@ -0,0 +1,206 @@
+		button.addActionListener(new Action());
+		message.addKeyListener(new Action());
+
+	}
+
+	/**
+	 * Start server and wait client to connect.
+	 */
+	public void serverClient() {
+
+		try {
+			server = new ServerSocket(9999);
+			client = server.accept();
+		} catch (IOException e) {
+
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * Method for recive message.
+	 */
+	public void reciveMessage() {
+
+		try {
+			BufferedReader reader = new BufferedReader(new InputStreamReader(
+					client.getInputStream()));
+			while (true) {
+				String str = reader.readLine();
+				conversation.append("Client:" + str + "\n");
+				someAction(str);
+
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * Method for send message.
+	 */
+	public void sendMessage() {
+		String str = message.getText();
+
+		try {
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+					client.getOutputStream()));
+
+			writer.write(str);
+			writer.newLine();
+			writer.flush();
+
+			conversation.append("Server:" + str + "\n");
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		message.setText("");
+	}
+
+	private class Action extends KeyAdapter implements ActionListener {
+
+		/**
+		 * Message is sent when the button is clicked.
+		 */
+
+		@Override
+		public void actionPerformed(ActionEvent e) {
+			if (e.getSource() == button) {
+				sendMessage();
+
+			}
+
+		}
+
+		/**
+		 * Message is sent when eneter key is clicked.
+		 */
+		@Override
+		public void keyPressed(KeyEvent e) {
+			if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+				sendMessage();
+			}
+		}
+
+	}
+
+	/**
+	 *If first word /web,web page will be open.
+	 *If first word /file, file will be open.
+	 *If first word / delete, fill will be delete.
+	 *If first word /list, client will recive a message with list of files.
+	 * 
+	 * @param message client send message.
+	 */
+	public void someAction(String message) {
+		try {
+			String[] str = message.split(" ", 2);
+			File file = new File(str[1]);
+
+			if (str[0].equals("/web")) {
+				Desktop.getDesktop()
+						.browse(new URL("http://" + str[1]).toURI());
+			} else if (str[0].equals("/open")) {
+				Desktop.getDesktop().open(file);
+			} else if (str[0].equals("/delete")) {
+				file.delete();
+			} else if (str[0].equals("/list")) {
+				String[] list = file.list();
+				BufferedWriter writer = new BufferedWriter(
+						new OutputStreamWriter(client.getOutputStream()));
+				writer.write("Files" + file);
+				writer.newLine();
+				for (String str1 : list) {
+					writer.write(str1);
+					writer.newLine();
+				}
+
+				writer.flush();
+
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (ArrayIndexOutOfBoundsException e) {
+			e.printStackTrace();
+		} catch (URISyntaxException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public static void main(String[] args) {
+
+		new Server();
+
+	}
+
```

Comment:
Danas sam do 5h imam poslije posla obaveza sutra sam tu cijeli dan :)

Ako hoces sutra kad ti bude odgovaralo?

On Aug 4, 2015, at 3:41 PM, Narena <notifications@github.com<mailto:notifications@github.com>> wrote:


In homework24Networking/src/ba/bitcamp/homework24Networking/Server.java<https://github.com/NarenaIbrisimovic/homeworks/pull/19#discussion_r36188230>:

> +                     e.printStackTrace();
> +             } catch (IOException e) {
> +                     e.printStackTrace();
> +             } catch (ArrayIndexOutOfBoundsException e) {
> +                     e.printStackTrace();
> +             } catch (URISyntaxException e) {
> +                     e.printStackTrace();
> +             }
> +     }
> +
> +     public static void main(String[] args) {
> +
> +             new Server();
> +
> +     }
> +


Cao Harise, Do kada si danas ili sutra u Atlantu da dodem imam pitanja :)? Ne znam kako da napravim da mi na jednom branchu bude samo odgovarajuca zadaca. LP, Narena
…<x-msg://30/#>
On Mon, Aug 3, 2015 at 12:26 PM, Haris Krajina <notifications@github.com<mailto:notifications@github.com>> wrote: In homework24Networking/src/ba/bitcamp/homework24Networking/Server.java <https://github.com/NarenaIbrisimovic/homeworks/pull/19#discussion_r36072883> : > + e.printStackTrace(); > + } catch (IOException e) { > + e.printStackTrace(); > + } catch (ArrayIndexOutOfBoundsException e) { > + e.printStackTrace(); > + } catch (URISyntaxException e) { > + e.printStackTrace(); > + } > + } > + > + public static void main(String[] args) { > + > + new Server(); > + > + } > + Ostalo je dobro, sredi exceptione i ove manje izmjene koje sam predlozio i moze merge [image: [:checkered_flag:] ] I jos jednom napomena da u pull requestu budu izmjene samo za datu zadacu, naravno bude sta nejasno javni skontamo zajedno. — Reply to this email directly or view it on GitHub <https://github.com/NarenaIbrisimovic/homeworks/pull/19/files#r36072883>.

—
Reply to this email directly or view it on GitHub<https://github.com/NarenaIbrisimovic/homeworks/pull/19/files#r36188230>.